api/v1alpha1/trafficmanagerprofile_types.go

```diff
@@ -0,0 +1,159 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,categories={fleet-networking},shortName=tmp
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.status.dnsName`,name="DNS-Name",type=string
+// +kubebuilder:printcolumn:JSONPath=`.status.conditions[?(@.type=='Programmed')].status`,name="Is-Programmed",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+
+// TrafficManagerProfile is used to manage a simple Azure Traffic Manager Profile using cloud native way.
+// https://learn.microsoft.com/en-us/azure/traffic-manager/traffic-manager-overview
+type TrafficManagerProfile struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of TrafficManagerProfile.
+	Spec TrafficManagerProfileSpec `json:"spec"`
+
+	// The observed status of TrafficManagerProfile.
+	// +optional
+	Status TrafficManagerProfileStatus `json:"status,omitempty"`
+}
+
+// TrafficManagerProfileSpec defines the desired state of TrafficManagerProfile.
+type TrafficManagerProfileSpec struct {
+	// The endpoint monitoring settings of the Traffic Manager profile.
+	// +optional
+	MonitorConfig *MonitorConfig `json:"monitorConfig,omitempty"`
```

Comment:
Are we going to use "Weighted routing method" always?

Reply:
yes, added comment there.